addons/mail/controllers/discuss.py

```diff
@@ -254,6 +254,7 @@ def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
             raise NotFound()
         if not request.env.user.share:
             # Check through standard access rights/rules for internal users.
+            attachment_sudo._attachment_delete_notification()
```

Comment:
I think this method should do both the notification and the following unlink(), and it should be our preferred way when we want to delete an attachment, without having to worry about calling both methods every time.